pycaret/time_series/forecasting/oop.py

```diff
@@ -2969,6 +2973,252 @@ def plot_model(
             return_obj = return_obj[0]
         return return_obj
 
+    def _predict_model_reconcile_pipe_estimator(
+        self, estimator: Union[BaseForecaster, PyCaretForecastingPipeline]
+    ) -> Tuple[PyCaretForecastingPipeline, BaseForecaster]:
+        """Returns the pipeline along with the final model in the pipeline.
+
+        # Use Cases:
+        # (1) User is in the middle of native experiment and passes a Base Model
+        #     without pipeline.
+        #     Action: Append pipeline and predict
+        # (2) User saved a model (without pipeline) and in the future, restarted
+        #     experiment after setup and loaded his model.
+        #     Action: Append pipeline and predict. If setup has not been run,
+        #     raise an exception.
+        # (3) User saved a model pipeline and in the future, loaded this pipeline
+        #     in another experiment to make predictions. This model pipeline might
+        #     be different from the experiment pipeline. Hence experiment pipeline
+        #     should not be changed. Predict as is.
+        #     Action: Pipeline as is
+
+        Parameters
+        ----------
+        estimator : Union[BaseForecaster, PyCaretForecastingPipeline]
+            Estimator passed by user
+
+        Returns
+        -------
+        Tuple[PyCaretForecastingPipeline, BaseForecaster]
+            The pipeline and the final model in the pipeline
+
+        Raises
+        ------
+        ValueError
+            When a model (without pipeline) is loaded into an experiment where
+            setup has not been run, but user wants to make a prediction.
+        """
+        if isinstance(estimator, PyCaretForecastingPipeline):
+            # Use Case 3
+            pipeline_with_model = deepcopy(estimator)
+            estimator_ = self._get_final_model_from_pipeline(
+                pipeline=pipeline_with_model, check_is_fitted=True
+            )
+        else:
+            if self._setup_ran:
+                # Use Case 1 & 2
+                # Deep Cloning to prevent overwriting the fh when user specifies their own fh
+                estimator_ = deepcopy(estimator)
+                pipeline_with_model = _add_model_to_pipeline(
+                    pipeline=self.pipeline, model=estimator_
+                )
+            else:
+                raise ValueError(
+                    "\n\nSetup has not been run and you have provided a estimator without the pipeline. "
```

Comment:
do we need the newlines here?

Reply:
It is easier to read the error messages. I will remove one new line (just keep one)